wolfssl/internal.h

```diff
@@ -2599,8 +2601,79 @@ typedef enum {
 #ifdef WOLFSSL_QUIC
     TLSX_KEY_QUIC_TP_PARAMS_DRAFT   = 0xffa5, /* from draft-ietf-quic-tls-27 */
 #endif
+#if defined(HAVE_ECH)
+    TLSX_ECH                        = 0xfe0d,
+#endif
 } TLSX_Type;
 
+#if defined(HAVE_ECH)
+
+typedef enum
+{
+    ECH_TYPE_OUTER = 0,
+    ECH_TYPE_INNER = 1
+} EchType;
+
+typedef enum
+{
+    ECH_WRITE_GREASE,
+    ECH_WRITE_REAL,
+    ECH_WRITE_RETRY_CONFIGS,
+    ECH_WRITE_NONE,
+    ECH_PARSED_INTERNAL,
+} EchState;
+
+typedef struct EchCipherSuite
+{
+    word16 kdfId;
+    word16 aeadId;
+} EchCipherSuite;
+
+typedef struct EchConfig
+{
+    byte configId;
+    word16 kemId;
+    void* receiverPrivkey;
+    byte receiverPubkey[HPKE_Npk_MAX];
+    byte numCipherSuites;
+    EchCipherSuite* cipherSuites;
+    char* publicName;
+    byte* raw;
+    word32 rawLen;
+    struct EchConfig* next;
+} EchConfig;
+
+typedef struct ECH
+{
+    byte type;
+    word16 kemId;
+    EchCipherSuite cipherSuite;
+    byte configId;
+    byte enc[HPKE_Npk_MAX];
+    word16 encLen;
+    word32 innerClientHelloLen;
+    word32 paddingLen;
+    byte* innerClientHello;
+    byte* outerClientPayload;
+    const byte* aad;
+    word32 aadLen;
+    Hpke* hpke;
+    void* ephemeralKey;
+    EchState state;
+    EchConfig* echConfig;
+} ECH;
```

Comment:
Concerned about possible naming collisions on `ECH`. Can you expand that? Also the items in the structure should be sorted by size. void*, word32's first, word16, byte, etc.